### ISSUE TITLE
TrainTestSplit without Label

### ISSUE THREAD

**Issue by keonkim** [mlpack/mlpack#651]
The current [TrainTestSplit](https://github.com/mlpack/mlpack/blob/master/src/mlpack/core/data/split_data.hpp#L102) only supports labeled dataset.
I think it would be good to provide the same TrainTestSplit functions for unlabeled datasets.

So The function signature would be: 
```
template<typename T>
std::tuple<arma::Mat<T>, arma::Mat<T>>
TrainTestSplit(const Mat<T>& input, const double testRatio )
```

**Comment by zoq** [comment]
Sounds good, in this case we can also split data where labels are the last row.

**Comment by rcurtin** [comment]
Agreed, we should also include the overload which returns `void` and takes in the training and test set matrices as a parameter.

**Comment by stereomatchingkiss** [comment]
Sounds good, I think std::paritition quite suit for this task. Do armadillo provide better function?

**Comment by keonkim** [comment]
@stereomatchingkiss I will try this, if you allow :)

**Comment by stereomatchingkiss** [comment]
@keonkim Then I will leave it to you, good luck

**Comment by keonkim** [comment]
This is resolved at #650. Closing this issue.
